community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -774,7 +773,17 @@ private void release()
      */
     private boolean canBeTerminated()
     {
-        return !closed && !terminated;
+        return !closed && !isTerminated();
+    }
+
+    private boolean isTerminated()
+    {
+        return terminationReason != null;
+    }
+
+    public long lastTransactionTimestampWhenStarted()
```

Comment:
This guy is not part of the `KernelTransaction` interface? If it is, it needs an `@Override`